### PR TITLE
Add relay selector to home feed

### PR DESCRIPTION
**Pull request by mplorentz**
## Issues covered
#1291

## Description
This adds the `RelayPicker` back to the home feed. When you select a relay from the picker the feed will show all events from the given relay.

I had to modify `PagedNoteDatasource` a little so that it can make its requests to the correct relays. I also polished up the design of RelayPicker a bit to more closely match Figma (I referenced [this design](https://www.figma.com/design/s0qf4VmyQygydP8MIQazZc/Nos?node-id=231-6141&t=7cdTEpT5nwnjKZoN-1) in addition to the one linked in the ticket).

I didn't add the "Relays" header to the picker or the note "Filtering by an individual relay will show you ALL the content on that relay.". I'd like to either do that in a separate PR or in #1292. I'll check with Linda on that.

## How to test
1. Open the Home tab
2. Tap the filter button in the top right
3. Choose a relay
4. Verify that the events from that relay are displayed.

## Screenshots/Video

|Before|After|
|----|-----|
|Simulator Screenshot - iPhone 15 - 2024-07-12 at 13 33 07|simulator_screenshot_855B29F1-C556-4A53-8739-E317BF68B840|

### CODE REVIEW

**Review by joshuatbrown — CHANGES_REQUESTED** [review]
Tested, too, and it looked good, though I know you're fixing an issue after merging from `main`.

File: Nos.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved
Comment (mplorentz) [reply Nos.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:219]: I think Martin is using a newer version of Xcode. I think I'd rather commit whatever the official version defined in `.xcode-version` says, which right now is 15.2. So I'd like to keep this I guess.

File: Nos/Controller/PagedNoteDataSource.swift
Comment (mplorentz) [reply Nos/Controller/PagedNoteDataSource.swift:14]: I called it that to emphasize that `relayFilter` and `databaseFilter` should return the same set of events. Or as close to the same as possible. If you saw `databaseFilter` in the code would that make you more likely to read the doc comments than `fetchRequest`? 😛

Comment (joshuatbrown) [reply Nos/Controller/PagedNoteDataSource.swift:14]: _Maybe_ 😀

Comment (joshuatbrown) [reply Nos/Controller/PagedNoteDataSource.swift:14]: If it was named `fetchRequestThatReturnsTheSameEventsAsRelayFilter` though, I wouldn't even have to read the docs.

**Comment by mplorentz** [comment]
When I merged from `main` it caused a new issue. The `oldestEventCreationDate`, which is integral to paging, was being stored in the `RelaySubscription` on `RelaySubscriptionManager`. But `RelaySubscriptionManager` is much more aggressive about closing and deallocating `RelaySubscriptions` that have finished returning events. 

This means that the `PagedRelaySubscription` needs to keep track of the `oldestEventCreationDate` itself. To do that it needs to get notified when the subscription processes new events. So I used Combine to create an observable list of events on `RelaySubscription` and gave `PagedRelaySubscription` a new piece of internal state, `oldestEventByRelay`, to keep track of the oldest event we have seen from each relay.

Since `RelaySubscription` is now hosting an observable stream of events it doesn't make sense to be copying it everywhere as a `struct`, so it's now a `class`. And now that things outside the `actor RelaySubscriptionManagerActor` are touching it I isolated `RelaySubscription` to the same actor as `RelaySubscriptionManagerActor`. 

I also discovered an existing bug with paging: if you filter the home feed by a single relay and scroll too fast sometimes it will stop paging. I will fix this somewhere else, but you might see it.

File: Nos/Views/Home/HomeFeedView.swift
Comment (mplorentz) [reply Nos/Views/Home/HomeFeedView.swift]: I tried this but unfortunately `.seconds()` takes an Int, and TimeInterval is a Double. So I guess that API isn't designed to work with TimeInterval :( I added "(in seconds)" to the docs though.

Comment (joshuatbrown) [reply Nos/Views/Home/HomeFeedView.swift]: Ah, I was looking above at the `+ Double(Self.staticLoadTime)` and thinking we could avoid that -- didn't see the `.seconds()` below. This makes sense, though I still wish it were TimeInterval and we just converted to an Int for `.seconds()`. It's a small wish, though, and not worth holding this PR. Also, I wish Apple could give us a `.seconds()` that accepts a `TimeInterval`...

**Review by joshuatbrown — APPROVED** [review]
Looks good and works for me!

File: Nos/Models/CoreData/Event+CoreDataClass.swift
Comment (mplorentz) [reply Nos/Models/CoreData/Event+CoreDataClass.swift:489]: Oops I forgot about this. Let's do it next time!

**Comment by mplorentz** [comment]
Alright changed to a `TimeInterval`

**Pull request merged**
